aii-ks/src/main/perl/ks.pm

```diff
@@ -1384,29 +1424,19 @@ end_of_yum_conf
 cat <<end_of_repos > /tmp/aii/yum/repos/aii.repo
 EOF
 
-    my ($phost, $pport, $ptype) = proxy($config);
-
-    $self->debug(5,"    Adding YUM repositories...");
+    $self->debug(5, "Adding YUM repositories...");
 
-    foreach my $repo (@$repos) {
-        if ($ptype && $ptype eq 'reverse') {
-            $repo->{protocols}->[0]->{url} =~ s{://[^/]*}{://$phost:$pport};
-        }
-        # mandatory in 16.4 schema
-        #   these values are the default values in the schema
-        $repo->{enabled} = 1 if(! defined($repo->{enabled}));
-        $repo->{gpgcheck} = 0 if(! defined($repo->{gpgcheck}));
+    foreach my $name (sort keys %$repos) {
+        my $repo = $repos->{$name};
 
         print <<EOF;
-[$repo->{name}]
-name=$repo->{name}
-baseurl=$repo->{protocols}->[0]->{url}
+[$name]
+name=$name
```

Comment:
I think this should remain `$repo->{name}`. Nothing requires the name attribute to be the same as the dictionary key was. I don't know if that distinction would matter in practice, though.

Reply:
what dictionary key? there is only one name, repositories in the `/software/repositories` are a list

Reply:
@gombasg  `get_repos` uses `$repo->{name}` as the key for the hash it returns, so I believe this is doing what you want.